example/advanced/server/advanced_server.cpp

```diff
@@ -581,79 +561,38 @@ class http_session : public std::enable_shared_from_this<http_session>
     run()
     {
         // Make sure we run on the strand
-        if(! strand_.running_in_this_thread())
+        if(! stream_.get_executor().running_in_this_thread())
```

Comment:
Isn't this condition always true?

Reply:
It is in theory true for the example but I want to make these components bullet-proof so if they are copied and pasted they will have a higher chance of working in adverse conditions. More can be done on this front.

Reply:
Because this is `io_context::strand` that condition may not be true - 2 strands may share implementations (`net::strand<E>` exhibits different behavior).